client/src/ui/track/newApplicationModal.tsx

```diff
@@ -0,0 +1,175 @@
+import React from "react";
+import { Modal, Typography, Button } from "@mui/material";
+import { InputField } from "@/ui/InputField"; 
+import { Searchbox } from "@/ui/track/Searchbox"; 
+import { faPenToSquare } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import UpdateStatus from "@/ui/track/UpdateStatus";
+// You will import your calendar component here once added
+
+interface NewApplicationModalProps {
+  open: boolean;
+  handleClose: () => void;
+}
+
+export default function NewApplicationModal({
+  open,
+  handleClose,
+}: NewApplicationModalProps) {
+  return (
+    <Modal
+      open={open}
+      onClose={handleClose}
+      sx={{
+        display: "flex",
+        alignItems: "center",
+        justifyContent: "center",
+      }}
+    >
+      <form
+        style={{
+          width: "70%", 
+          padding: "30px 40px", 
+          backgroundColor: "white", 
+          borderRadius: "8px",
+          display: "flex",
+          flexDirection: "column",
+          gap: "30px",
+        }}
+      >
+        <Typography
+          variant="h4"
+          style={{
+            display: "flex",
+            alignItems: "center",
+            marginBottom: "10px",
+          }}
+        >
+          <FontAwesomeIcon
+            icon={faPenToSquare}
+            style={{ marginRight: "10px" }}
+          />
+          Add New Application
+        </Typography>
+
+        {/* Container for the status dropdown and input fields */}
+        <div
+          style={{
+            display: "flex",
+            flexDirection: "column",
+            gap: "20px",
+          }}
+        >
+          {/* Status dropdown */}
+          <div style={{ width: "150px" }}>
+            <UpdateStatus
+              required
+              name="status"
+              options={["Applied", "OA", "Interviewing", "Offer", "Rejected"]}
+            />
+          </div>
+
+          {/* Input Fields in a single flex container */}
+          <div style={{ display: "flex", gap: "40px" }}>
+            {/* Left column */}
+            <div style={{ flex: 1 }}>
+              <InputField
+                label="Role Title"
+                placeholder="Title"
+                name="roleTitle" 
+                required
+                fullWidth
+                sx={{ marginBottom: "20px" }}
+              />
+              <InputField
+                label="Location"
+                placeholder="Location"
+                name="location" // Added name prop
+                fullWidth
+                sx={{ marginBottom: "20px" }}
+              />
+
+              {/* Calendar component for Date Applied */}
+              {/* Replace this with the actual calendar component */}
```

Comment:
nit - Add 'TODO' on this. It makes it easier to search later for code that needs to get updated later

Reply:
Done!